service/storage_service.cc

```diff
@@ -2652,8 +2652,10 @@ class join_node_rpc_handshaker : public service::group0_handshaker {
         _ss._join_node_group0_started.set_value();
 
         // Processing of the response is done in `join_node_response_handler`.
-        // Wait for it to complete.
-        co_await _ss._join_node_response_done.get_shared_future(lowres_clock::now() + std::chrono::minutes(3));
+        // Wait for it to complete. If the topology coordinator fails to
+        // deliver the rejection, it won't complete. In such a case, the
+        // operator is responsible for shutting down the joining node.
+        co_await _ss._join_node_response_done.get_shared_future();
```

Comment:
Well, is it possible to shutdown the node now gracefully -- since we don't provide a reference to any abort_source?

Reply:
Right, we need to pass `abort_source` there. There is even a TODO in `raft_group0::join_group0`:
```cpp
// TODO: aborts?
if (co_await handshaker->post_server_start(g0_info)) {
    break;
}
```
I'll fix it.

Reply:
Done.